src/main/java/mate/academy/impl/AbstractDao.java

```diff
@@ -0,0 +1,99 @@
+package mate.academy.impl;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.exception.DataProcessingException;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+
+public abstract class AbstractDao<T, I extends Serializable> {
```

Comment:
you already have AbstractDao, RoleDaoImpl, UserDaoImpl in the mate.academy.dao.impl. Why did you add it?